djstripe/models.py

```diff
@@ -189,12 +190,19 @@ def process(self):
                     if not self.customer:
                         self.link_customer()
                     if self.customer:
+                        # TODO see whether sync_subscriptions() also covers the deletion case.
+                        """
+                        Previously, the code tested for "customer.subscription.deleted" and in
+                        this case would cancel the current_subscription, rather than syncing.
+                        
                         if self.kind == "customer.subscription.deleted":
                             self.customer.current_subscription.status = CurrentSubscription.STATUS_CANCELLED
                             self.customer.current_subscription.canceled_at = timezone.now()
                             self.customer.current_subscription.save()
                         else:
                             self.customer.sync_current_subscription()
+                        """
```

Comment:
Yes, I see the issue. The new Customer.sync_subscriptions() needs to be augmented to check for subscriptions which no longer have matching counterparts at the Stripe end. At the moment it only loops through each Stripe subscription and tries to match it up with its dj-stripe counterpart.